src/backend/distributed/utils/metadata_cache.c

```diff
@@ -3415,3 +3419,26 @@ DistNodeMetadata(void)
 
 	return metadata;
 }
+
+
+Datum
+role_exists(PG_FUNCTION_ARGS)
+{
+	Name roleName = PG_GETARG_NAME(0);
+	bool roleExists = SearchSysCacheExists1(AUTHNAME, NameGetDatum(roleName));
+
+	PG_RETURN_BOOL(roleExists);
+}
+
+
+Datum
+authinfo_valid(PG_FUNCTION_ARGS)
+{
+	ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+					errmsg("catalog table unsupported"),
```

Comment:
Maybe `cannot write to pg_dist_authinfo`. That's a little clearer in case this call happens somewhere deep down.

Reply:
Sure.